internal/provider/resource_string.go

```diff
@@ -663,98 +640,84 @@ func stringSchemaV1() tfsdk.Schema {
+				Optional:    true,
 			},
 
-			"length": {
+			"length": schema.Int64Attribute{
 				Description: "The length of the string desired. The minimum value for length is 1 and, length " +
 					"must also be >= (`min_upper` + `min_lower` + `min_numeric` + `min_special`).",
-				Type:     types.Int64Type,
 				Required: true,
 			},
 
-			"special": {
+			"special": schema.BoolAttribute{
 				Description: "Include special characters in the result. These are `!@#$%&*()-_=+[]{}<>:?`. Default value is `true`.",
-				Type:        types.BoolType,
 				Optional:    true,
 				Computed:    true,
 			},
 
-			"upper": {
+			"upper": schema.BoolAttribute{
 				Description: "Include uppercase alphabet characters in the result. Default value is `true`.",
-				Type:        types.BoolType,
 				Optional:    true,
 				Computed:    true,
 			},
 
-			"lower": {
+			"lower": schema.BoolAttribute{
 				Description: "Include lowercase alphabet characters in the result. Default value is `true`.",
-				Type:        types.BoolType,
 				Optional:    true,
 				Computed:    true,
 			},
 
-			"number": {
+			"number": schema.BoolAttribute{
 				Description: "Include numeric characters in the result. Default value is `true`.",
-				Type:        types.BoolType,
 				Optional:    true,
 				Computed:    true,
 			},
-
-			"min_numeric": {
+			"min_numeric": schema.Int64Attribute{
 				Description: "Minimum number of numeric characters in the result. Default value is `0`.",
-				Type:        types.Int64Type,
 				Optional:    true,
 				Computed:    true,
 			},
 
-			"min_upper": {
+			"min_upper": schema.Int64Attribute{
 				Description: "Minimum number of uppercase alphabet characters in the result. Default value is `0`.",
-				Type:        types.Int64Type,
 				Optional:    true,
 				Computed:    true,
 			},
 
-			"min_lower": {
+			"min_lower": schema.Int64Attribute{
 				Description: "Minimum number of lowercase alphabet characters in the result. Default value is `0`.",
-				Type:        types.Int64Type,
 				Optional:    true,
 				Computed:    true,
 			},
 
-			"min_special": {
+			"min_special": schema.Int64Attribute{
 				Description: "Minimum number of special characters in the result. Default value is `0`.",
-				Type:        types.Int64Type,
 				Optional:    true,
 				Computed:    true,
 			},
 
-			"override_special": {
+			"override_special": schema.StringAttribute{
 				Description: "Supply your own list of special characters to use for string generation.  This " +
 					"overrides the default character list in the special argument.  The `special` argument must " +
 					"still be set to true for any overwritten characters to be used in generation.",
-				Type:     types.StringType,
 				Optional: true,
 				Computed: true,
 			},
 
-			"result": {
+			"result": schema.StringAttribute{
 				Description: "The generated random string.",
-				Type:        types.StringType,
 				Computed:    true,
+				Sensitive:   true,
```

Comment:
Not sure if this was copied from anywhere, but FYI on this addition that doesn't need to be here.